### PR TITLE
Bug 1346582 - Disable toolbar home button for internal URLs

### PR DESCRIPTION
**Pull request by mauryat**

### CODE REVIEW

File: Client/Frontend/Accessors/HomePageAccessors.swift [hunk 1]
```diff
@@ -26,8 +26,14 @@ class HomePageAccessors {
...
         switch state.ui {
-        case .tab:
-            return true
+        case .tab(let tabState):
+            if hasHomePage(state) {
+                return true
+            } else if let url = tabState.url {
+                return url.isValid
+            } else {
+                return false
```
Comment (thebnich) [inline Client/Frontend/Accessors/HomePageAccessors.swift]: You can remove the `else` conditions here and shorten this to `return tabState.url?.isValid ?? false`.

File: Shared/Extensions/NSURLExtensions.swift [hunk 2]
```diff
@@ -289,6 +289,10 @@ extension URL {
         return host.lowercased() == "localhost" || host == "127.0.0.1"
     }
 
+    public var isValid: Bool {
+        return isWebPage(includeDataURIs: true) && !isLocal
+    }
```
Comment (thebnich) [inline Shared/Extensions/NSURLExtensions.swift]: I'd like to avoid creating another extension on `URL` if possible. Isn't `isValid` equivalent to `!isLocal`?

Comment (thebnich) [inline Shared/Extensions/NSURLExtensions.swift]: I suppose there are cases where `isValid` would false return where `!isLocal` would not. I guess we can keep the extension here, but let's name it something more descriptive (local URLs are still valid URLs). How about `isRemoteWebPage`?

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
AppMenuConfiguration.swift | 311 | Limit vertical whitespace to a single empty line. Currently 2. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by garvankeeley** [comment]
What is an internal url, do we mean about:home/license etc?
(trying to take action on this PR because it has sat for a while)

**Comment by mauryat** [comment]
@garvankeeley, this is ready for merge. Just waiting on a review.

**Comment by garvankeeley** [comment]
thanks, i'll assign it to myself

**Comment by garvankeeley** [comment]
I applied this test in this patch XCUITests/HomePageUITest.swift to master, in order to see the current bug, however the tests pass, suggesting this was fixed on master. Can you confirm this fixed already on master?
